src/main/java/org/broadinstitute/hellbender/tools/spark/sv/FindBadGenomicKmersSpark.java

```diff
@@ -31,7 +32,9 @@
     private static final long serialVersionUID = 1L;
     @VisibleForTesting static final Long MAX_KMER_FREQ = 3L;
     private static final int REF_RECORD_LEN = 10000;
-    private static final int CHUNK_SIZE = 3000000;
+    // to give a task size of 100Kb -- the max size recommended in the Spark documentation
+    private static final int REF_RECORDS_PER_PARTITION = 102400 / REF_RECORD_LEN;
```

Comment:
Not sure it's worth re-opening for this tool since it's a run-once on the
reference, but I was working on an implementation that followed Tom's
approach in the branch cw_clusterize_sv_spark_tools. That implementation
takes around 35 minutes on the cluster to run in its current state.

On Tue, Feb 23, 2016 at 12:11 PM, Tom White <notifications@github.com>
wrote:

> In
> src/main/java/org/broadinstitute/hellbender/tools/spark/sv/FindBadGenomicKmersSpark.java
> <https://github.com/broadinstitute/gatk/pull/1505#discussion_r53813728>:
>
> > @@ -31,7 +32,9 @@
> >      private static final long serialVersionUID = 1L;
> >      @VisibleForTesting static final Long MAX_KMER_FREQ = 3L;
> >      private static final int REF_RECORD_LEN = 10000;
> > -    private static final int CHUNK_SIZE = 3000000;
> > +    // to give a task size of 100Kb -- the max size recommended in the Spark documentation
> > +    private static final int REF_RECORDS_PER_PARTITION = 102400 / REF_RECORD_LEN;
>
> Sorry to come to this late. The error message is because the size of the
> serialized task is more than 100KB, not because the partition size is
> larger than that size. I would recommend avoiding using parallelize in
> this way, since everything is loaded into memory on one machine then sent
> across the network to run (and the task sizes will be very small due to the
> 100KB size).
>
> What I would do is broadcast the reference and have each task process a
> part of it. Maybe parallelize a list of SimpleIntervals and then each task
> can work on one of them. See BroadcastJoinReadsWithRefBases for how we
> broadcast a ReferenceMultiSource.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/broadinstitute/gatk/pull/1505/files#r53813728>.
>